### PR TITLE
Add tests brakeman rb

### PR DESCRIPTION
**Pull request by bethanyr**
I have added a couple of tests for the brakeman.rb file to increase code coverage for issue #723. I plan to add more tests, but wanted to get feedback and make sure I am following the current pattern for testing.

### CODE REVIEW

**Comment by presidentbeef** [comment]
Hi Bethany,

Thank you for your work on this. The style is fine. However, unless I'm missing something, you shouldn't need to create a config file for the tests you added. That was in the other tests because they were specifically testing behavior when reading config files.

Otherwise, great!

File: test/tests/brakeman.rb [hunk 1]
```diff
@@ -278,6 +278,49 @@ def test_output_format
...
     output_format_tester({:output_files => ['xx.html', 'xx.pdf', 'xx.csv', 'xx.tabs', 'xx.json', 'xx.md']}, [:to_html, :to_pdf, :to_csv, :to_tabs, :to_json, :to_markdown])
   end
+
+  def test_output_format_errors_raised
+    options = {:output_format => :to_json, :output_files => ['xx.csv', 'xx.xxx']}
+    assert_raises("ArgumentError: Cannot specify output format if multiple output files specified") {Brakeman.get_output_formats(options)}
+  end
+
+  def test_github_options_raises_error
+    config = Tempfile.new("config")
```
Comment (presidentbeef) [inline test/tests/brakeman.rb]: No need for the config file.

File: test/tests/brakeman.rb [hunk 2]
```diff
@@ -278,6 +278,49 @@ def test_output_format
...
...t escape single quotes (CVE-2012-3464)\nCheckSkipBeforeFilter         Warn when skipping CSRF or authentication checks by default\nCheckSQL                      Check for SQL injection\nCheckSQLCVEs                  Checks for several SQL CVEs\nCheckSSLVerify                Checks for OpenSSL::SSL::VERIFY_NONE\nCheckStripTags                Report strip_tags vulnerabilities CVE-2011-2931 and CVE-2012-3465\nCheckSymbolDoSCVE             Checks for versions with ActiveRecord symbol denial of service vulnerability\nCheckTranslateBug             Report XSS vulnerability in translate helper\nCheckUnsafeReflection         Checks for unsafe reflection\nCheckValidationRegex          Report uses of validates_format_of with improper anchors\nCheckWithoutProtection        Check for mass assignment using without_protection\nCheckXMLDoS                   Checks for XML denial of service (CVE-2015-3227)\nCheckYAMLParsing              Checks for YAML parsing vulnerabilities (CVE-2013-0156)\nCheckSymbolDoS                Checks for symbol denial of service\nCheckUnscopedFind             Check for unscoped ActiveRecord queries\nCheckWeakHash                 Checks for use of weak hashes like MD5\n"
```
Comment (presidentbeef) [inline test/tests/brakeman.rb]: I'm worried about maintaining this. The list of checks is only going to grow.

What about something like this (untested)?

```ruby
check_list = capture_output { Brakeman.list_checks(options) }
Brakeman::Checks.checks.each do |check|
  assert check_list.include? check.class.to_s
  assert check_list.include? check.description
end
```

Same with the optional checks.

Comment (bethanyr) [inline test/tests/brakeman.rb]: good suggestion. I have updated the 2 tests to get the list of checks from Brakeman::Checks. There is currently a private method: get_check_name that is in Brakeman::Checks - which has the same code: to_s.split("::").last that I am using in the test. Do you have any objections if I make the method public to call it from the test? Or is the test code OK?

File: test/tests/brakeman.rb (hunk 1 above)
Comment (bethanyr) [inline test/tests/brakeman.rb]: good catch - I've removed it

File: test/tests/brakeman.rb (hunk 2 above)
Comment (presidentbeef) [inline test/tests/brakeman.rb]: You can make it public.

File: test/tests/brakeman.rb [hunk 3]
```diff
@@ -278,6 +278,75 @@ def test_output_format
...
+      assert check_list.include? check.description
+    end
+  end
+
+  def test_dump_config_no_file
+    options = {:create_config => true, :test_option => "test"}
+    config_output = capture_output {
+      Brakeman.dump_config(options)
+    }[1]
+    assert_equal config_output, "---\n:test_option: test\n"
```
Comment (presidentbeef) [inline test/tests/brakeman.rb]: It would be even cooler if this loaded the YAML and compared it to the original `options`

**Comment by presidentbeef** [comment]
Hi Bethany, still working on this or ready to merge?

**Comment by bethanyr** [comment]
Hi Justin - I think it's ready to merge. Currently lib/brakeman.rb is at 88% coverage.

**Comment by presidentbeef** [comment]
Thanks! That put us up over 90%!

**Pull request merged**